src/code/z_parameter.c

```diff
@@ -52,38 +332,750 @@
+                if (gSaveContext.save.inventory.items[temp + i] == ITEM_BOTTLE) {
+                    return ITEM_NONE;
+                }
+            }
+        } else {
+            for (i = 0; i < 6; i++) {
+                if (gSaveContext.save.inventory.items[temp + i] == ITEM_NONE) {
+                    return ITEM_NONE;
+                }
+            }
+        }
+    } else if ((item >= ITEM_MOON_TEAR) && (item <= ITEM_MASK_GIANT)) {
+        return ITEM_NONE;
+    }
+
+    return gSaveContext.save.inventory.items[slot];
+}
+
+u8 Item_CheckObtainability(u8 item) {
+    return Item_CheckObtainabilityImpl(item);
+}
+
+void Inventory_DeleteItem(s16 item, s16 slot) {
+    s16 btn;
+
+    gSaveContext.save.inventory.items[slot] = ITEM_NONE;
+
+    for (btn = 1; btn < 4; btn++) {
+        if (GET_CUR_FORM_BTN_ITEM(btn) == item) {
+            SET_CUR_FORM_BTN_ITEM(btn, ITEM_NONE);
+            SET_CUR_FORM_BTN_SLOT(btn, SLOT_NONE);
+        }
+    }
+}
+
+void Inventory_UnequipItem(s16 item) {
+    s16 btn;
+
+    for (btn = 1; btn < 4; btn++) {
+        if (GET_CUR_FORM_BTN_ITEM(btn) == item) {
+            SET_CUR_FORM_BTN_ITEM(btn, ITEM_NONE);
+            SET_CUR_FORM_BTN_SLOT(btn, SLOT_NONE);
+        }
+    }
+}
+
+s32 Inventory_ReplaceItem(GlobalContext* globalCtx, u8 oldItem, u8 newItem) {
+    u8 i;
+
+    for (i = 0; i < 24; i++) {
+        if (gSaveContext.save.inventory.items[i] == oldItem) {
+            gSaveContext.save.inventory.items[i] = newItem;
+
+            for (i = 1; i < 4; i++) {
+                if (GET_CUR_FORM_BTN_ITEM(i) == oldItem) {
+                    SET_CUR_FORM_BTN_ITEM(i, newItem);
+                    Interface_LoadItemIconImpl(globalCtx, i);
+                    break;
+                }
+            }
+            return true;
+        }
+    }
+    return false;
+}
+
+void Inventory_UpdateDeitySwordEquip(GlobalContext* globalCtx) {
+    InterfaceContext* interfaceCtx = &globalCtx->interfaceCtx;
+    u8 btn;
+
+    if (CUR_FORM == PLAYER_FORM_FIERCE_DEITY) {
+        interfaceCtx->unk_21C = 0;
+        interfaceCtx->unk_21E = 0;
+        if ((((gSaveContext.save.playerForm > 0) && (gSaveContext.save.playerForm < 4))
+                 ? 1
+                 : gSaveContext.save.playerForm >> 1) == 0) {
+            BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) = ITEM_SWORD_DEITY;
+        } else {
+            if (BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) == ITEM_SWORD_DEITY) {
+                if (GET_CUR_EQUIP_VALUE(EQUIP_SWORD) == 0) {
+                    BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) = ITEM_NONE;
+                } else {
+                    BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) =
+                        GET_CUR_EQUIP_VALUE(EQUIP_SWORD) + ITEM_SWORD_KOKIRI - 1;
+                }
+            }
+        }
+    }
+
+    for (btn = 0; btn < 1; btn++) {
+        if ((GET_CUR_FORM_BTN_ITEM(btn) != ITEM_NONE) && (GET_CUR_FORM_BTN_ITEM(btn) != ITEM_UNK_FD)) {
```

Comment:
An amusing coincidence, but I suppose this means we know what it is?

Reply:
You mean `ITEM_UNK_FD`? I'm still not entirely sure

Reply:
Hmm, maybe not, never mind.